vendor/plugins/edu.harvard.dce.paella.timedCommentsPlugin/timedCommentsOverlayPlugin.js

```diff
@@ -0,0 +1,714 @@
+  },
+
+  onEvent: function (eventType, params) {
+    var thisClass = this;
+    switch (eventType) {
+
+      case paella.events.play:
+      // play means focus is off the comment box so it's ok to scroll
+      thisClass._isAutoScroll = true;
+      thisClass.updateCurrentTimeStamp();
+      if (thisClass._isActive) {
+        thisClass.scrollTimedComments(thisClass._isAutoScroll);
+      }
+      break;
+
+      case paella.events.timeupdate:
+      thisClass.updateCurrentTimeStamp();
+      if (thisClass._isActive) {
+        thisClass.scrollTimedComments(thisClass._isAutoScroll);
+      }
+      break;
+
+      case paella.events.pause:
+      case paella.events.endVideo:
+      thisClass._isAutoScroll = false;
+      break;
+
+      case paella.events.showTimedComments:
+      thisClass.loadTimedComments();
+      if (paella.player.playing()) {
+        thisClass._isAutoScroll = true;
+      }
+      break;
+
+      case paella.events.hideTimedComments:
+      thisClass._isActive = false;
+      if (thisClass._rootElement) {
+        thisClass.unloadTimedComments();
+      }
+      break;
+
+      case paella.events.refreshTimedComments:
+      if (thisClass._isActive) {
+        thisClass.reloadComments(params.data);
+      }
+    }
+    thisClass.lastEvent = eventType;
+  },
+
+  reloadComments: function (annotData) {
+    var thisClass = this;
+    thisClass._curScrollTop = $("#innerAnnotation") ? $("#innerAnnotation").scrollTop(): 0;
+    // isActive is set back to true in data load promise
+    thisClass._isActive = false;
+    if (thisClass._rootElement) {
+      $(thisClass._rootElement).empty();
+      $(thisClass._rootElement).resizable('destroy');
+      $(thisClass._rootElement).draggable('destroy');
+    }
+    thisClass.loadTimedComments(annotData);
+  },
+
+  unloadTimedComments: function () {
+    var thisClass = this;
+    if (thisClass._rootElement) {
+      $(thisClass._rootElement).remove();
+    }
+  },
+
+  loadTimedComments: function (annotData) {
+    var thisClass = this;
+    if (annotData) {
+      thisClass.loadWithData(annotData);
+    } else {
+      paella.data.read('timedComments', {
+        id: paella.initDelegate.getId()
+      },
+      function (data, status) {
+        thisClass.loadWithData(data);
+      });
+    }
+  },
+
+  loadWithData: function (data) {
+    var thisClass = this;
+    thisClass._annotations = data;
+    thisClass.sortAnnotations();
+    thisClass.drawTimedComments();
+
+    $("#innerAnnotation").animate({
+      scrollTop: thisClass._curScrollTop
+    },
+    100);
+
+    // changing the layout profile that is most optimal to show comments
+    thisClass.changeToOptimalVideoProfile(thisClass._optimalProfile);
+    thisClass._isActive = true;
+  },
+
+  // This is having problems in multiple browsers and dates of concurrent comments are not correct
```

Comment:
I have to say that I prefer the previous version of the sortAnnotations function:
sortAnnotations: function () {
    var thisClass = this;
    if (thisClass._annotations) {
      // First sort by inpoint, then by 'comment' id, then by annotation date
      thisClass._annotations = thisClass._annotations.sort(function (a, b) {
        base.log.debug("RS annotationId, a: " + a.annotationId + ", b: " + b.annotationId);
        base.log.debug("RS inpoint, a: " + a.inpoint + ", b: " + b.inpoint);

        // First, sort by inpoint (a comment and its replies will have the same inpoint)
        var ret = a.inpoint - b.inpoint;
        if (ret != 0)
          return ret;
        // Second, sort by parent comment id (doesn't matter which comes first, this is just to group comments and their replies)
        var aTimedComment = a.value.timedComment;
        var bTimedComment = b.value.timedComment;
        var aCommentId = aTimedComment.mode == "comment" ? aTimedComment.id : aTimedComment.parent;
        var bCommentId = bTimedComment.mode == "comment" ? bTimedComment.id : bTimedComment.parent;
        base.log.debug("RS commentId, a: " + aCommentId + ", b: " + bCommentId);

        ret = String(aCommentId).localeCompare(String(bCommentId));
        if (ret != 0)
          return ret;
        // Then, sort by annotation date (a comment will be created earlier than its replies)
        base.log.debug("RS created date, a: " + a.created + ", b: " + b.created);

        return new Date(a.created).getMilliseconds() - new Date(b.created).getMilliseconds();
      });
    }
  },